Core/clim-basic/utils.lisp

```diff
@@ -808,161 +808,76 @@ a flag CLOSED is T then beginning and end of the list are consecutive too."
+        finally
+           (if (null opportunity)
+               (return (values limit :emergency))
+               (return (values opportunity :soft)))))
+
+(defun line-breaks (string width
+                    &key
+                      (break-strategy t)
+                      initial-offset
+                      (margin nil margin-p)
+                      (start 0) end count
+                    &aux
+                      (width-fn (etypecase width
+                                  (function width)
+                                  (number (lambda (string start end)
+                                            (declare (ignore string))
+                                            (* width (- end start))))))
+                      (width (etypecase width
+                               (function nil)
+                               (number width)))
+                      (break-fn (etypecase break-strategy
+                                  (function
+                                   break-strategy)
+                                  ;; break everywhere
+                                  ((eql nil)
+                                   (lambda (string start index)
+                                     (declare (ignore start))
+                                     (case (char string index)
+                                       (#\newline :hard)
+                                       (otherwise :soft))))
+                                  ;; default strategy
+                                  ((eql t)
+                                   (lambda (string start index)
+                                     (declare (ignore start))
+                                     (case (char string index)
+                                       (#\newline :hard)
+                                       (#\space   :soft)
+                                       (otherwise nil))))
+                                  ;; In case of sequences we assume
+                                  ;; that the string is a single line
+                                  ;; without any hard line breaks
+                                  (list
+                                   (lambda (string start index)
+                                     (declare (ignore start))
+                                     (when (member (char string index)
+                                                   break-strategy)
+                                       :soft)))
+                                  (vector
+                                   (lambda (string start index)
+                                     (declare (ignore start string))
+                                     (when (find index break-strategy)
+                                       :soft)))))
+                      (initial-offset (or initial-offset 0))
+                      (margin (or margin
+                                  (* 80 (funcall width-fn "m" 0 1))))
+                      (start (or start 0))
```

Comment:
I think this is going way too far. It separates the lambda list from the documentation and the invariants at the beginning of the body by more than a screen.

The computation for `break-fn` might warrant helper function in any case.

Reply:
OK, I've moved it to a separate function.